templates/code_generation_swift_file.hbs

```diff
@@ -36,27 +36,44 @@ public enum LocaliciousQuantity: String {
 
 public extension LocalizationProvider {
     var translation: String? {
-        return translationKey.map(translation) ?? nil
+        return translationKey.map { translation(forKey: $0) } ?? nil
     }
 
     var accessibilityHint: String? {
-        return accessibilityHintKey.map(translation) ?? nil
+        return accessibilityHintKey.map { translation(forKey: $0) } ?? nil
     }
 
     var accessibilityLabel: String? {
-        return accessibilityLabelKey.map(translation) ?? nil
+        return accessibilityLabelKey.map { translation(forKey: $0) } ?? nil
     }
 
     var accessibilityValue: String? {
-        return accessibilityValueKey.map(translation) ?? nil
+        return accessibilityValueKey.map { translation(forKey: $0) } ?? nil
     }
 
-    private func translation(forKey key: String) -> String? {
-        let translation = NSLocalizedString(key, comment: "")
-        if let translationArgs = translationArgs {
-            return String(format: translation, arguments: translationArgs)
+    func translation(forKey key: String, withBundle bundle: Bundle = Bundle.main) -> String? {
+        let value = bundle.localizedString(forKey: key, value: nil, table: nil)
+
+        guard value != key else {
+            return nil
+        }
+
+        let translationArgs = self.translationArgs ?? []
+        guard translationArgs.count > 0 else { return value }
+        let format = value
+        let placeholdersCount = placeholders(in: format)
+        guard translationArgs.count == placeholdersCount else {
+            return nil
+        }
+
+        return String(format: format, arguments: translationArgs)
+    }
+
+    private func placeholders(in string: String) -> Int {
+        let supportedSeparators = ["$@", "$d"]
```

Comment:
This is going to be a bit tricky if we add additional placeholders. Good to go for now, but we need to think of a better solution here, that maybe even guarantees compile-time correctness?

Reply:
It would be nice to have instead of 
`public static func Title(args: CVarArg...) -> LocaliciousData`

`public static func Title(arg1: Int, arg2: String) -> LocaliciousData`

Based on the number of placeholders and their types.

But I don't know if we can support this in the template generation.

Reply:
But then how can I get all the args of the function to init the String?
It would require that to any compile-time method, before calling translation, we should come up with an array of all the parameters to use in the String initializer.
If we can find this "reflection" to concatenate all typed parameters, isn't it the same as `CVarArgList` with extra steps?

The difference that I see is type safety in the outer layer only

Reply:
It is indeed not possible to prevent runtime errors when calling `String(format: format, arguments: translationArgs)`. But if we provide a type-safe outer layer, users of Localicious can use this type-safe layer only and hence never run into a runtime issue?

Reply:
Yes, it would be the most fantastic option! No mistakes ever again, and you can know how parameters you have just by using the copy 🚀 

The things that I think that we need to implement are:
- Count the placeholders (differentiating them by their position, first, last or middle)
- Identify every parameter with the format type (string or int)
- Instead of using `containsFormatting`, https://github.com/PicnicSupermarket/localicious/blob/308d7cd06ba10271271efa13d53810dd977b5b6e/templates/code_generation_swift_child.hbs#L17 render a template of parameters
- Store the parameters
- Implement `toCVarArg` as part of the code generation based on the number of parameters and their correct order
- Call `toCVarArg` when translating

Does it make sense?

Reply:
Yes, makes sense! Let's tackle that in a separate PR.